library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
I believe the correct instance would be
```lean
meta instance : has_quote pexpr := ⟨λ e, `(%%e)⟩
```
We need a `pexpr` at tactic parse time that will evaluate to a `pexpr` at tactic run time. I _think_ this instance should just construct `quote_macro(e)` -- a Lean function for printing the raw structure of a `pexpr/expr` would be very helpful. Also, now my head hurts.

> BTW, I think I found another problem with open terms (but I'm completely fine with ignoring it):

Yes, we should just flag an error:

> I think we may assume (and check) that both parameters to parse will always be closed terms. I don't see any use in parameter parsers that can depend on the values of previous parameters.